homeworks/week4/gen_random.py

```diff
@@ -0,0 +1,53 @@
+import random
+class Segment:
+    def __init__(self, a, b):
+        self.start = a
+        self.end = b
+    def length(self):
+        return self.end - self.start
+    def __str__(self):
+        return "[" + str(self.start) + ", " + str(self.end) + "]"
+
+#Two-step randomizing: 1) Choose interval, based on its length. 2) Uniformly random a point on chosen interval
+#Complexity: O(logN) since random.choice use binary search on cumulative weights for searching the index
+#Auxiliary space: O(N) 
+def generateRandomValue(intervals, weights):
+    chosen = random.choices(intervals, cum_weights = weights, k = 1)[0]
```

Comment:
This is a really nice short solution!

But in an interview, interviewers will ask interviewee to not using build-in library like this one :). I made a note on the problem to make it more clear.

Yes, random.choices uses bisect - binary search.
https://github.com/python/cpython/blob/master/Lib/random.py#L497